ethereum/contracts/pyth/PythGovernance.sol

```diff
@@ -67,13 +70,33 @@ abstract contract PythGovernance is PythGetters, PythSetters, PythGovernanceInst
 
     function upgradeUpgradableContract(UpgradeContractPayload memory payload) virtual internal;
 
-    function setGovernanceDataSource(bytes memory encodedPayload) internal {
-        SetGovernanceDataSourcePayload memory payload = parseSetGovernanceDataSourcePayload(encodedPayload);
-
+    // Transfer the governance data source to a new value with sanity checks
+    // to ensure the new governance data source can manage the contract.
+    function transferGovernanceDataSource(bytes memory encodedPayload) internal {
         PythInternalStructs.DataSource memory oldGovernanceDatSource = governanceDataSource();
 
-        setGovernanceDataSource(payload.newGovernanceDataSource);
-        setLastExecutedGovernanceSequence(payload.initialSequence);
+        TransferGovernanceDataSourcePayload memory payload = parseTransferGovernanceDataSourcePayload(encodedPayload);
+
+        // Make sure the VAA is valid
```

Comment:
Could we make it clear in this comment that this is a second, claim, VAA? Maybe even extract everything to do with the second VAA into it's own function with a docstring.

Reply:
Worth re-iterating here that the purpose of ensuring the claim VAA is valid is to ensure we can emit governance messages from the new source.

Reply:
I don't like extracting it out to another function. The vaa check is the only logic inside this function and it doesn't make sense to add another one.